pegasus/sites.v3/code.org/public/educate/it.md

```diff
@@ -37,7 +37,7 @@ For the very best experience with all Code.org content, we recommend consulting
 | **To use YouTube hosted videos** <br/>(Deprecated late July 2018)                        | `https://s.youtube.com/*`<br/>`https://www.youtube.com/*`<br/>`https://*.googlevideo.com/*`<br/>`https://*.ytimg.com/*`                                |
 | **To use Code.org hosted videos**                                                        | **Unblock:**<br/>`https://videos.code.org`<br/>**Block:**<br/>`https://www.youtube.com/favicon.ico`<br/>`https://www.youtube-nocookie.com/favicon.ico` |
 | **To use Internet Simulator**                                                            | `https://api.pusherapp.com`<br/>`wss://ws.pusherapp.com:443`                                                                                           |
-| **To use Web Lab**                                                                       | `https://downloads.computinginthecore.org`<br/>`https://codeprojects.org`                                                                              |
+| **To use Web Lab**                                                                       | `https://*.codeprojects.org`                                                                              |
```

Comment:
@code-dot-org/infrastructure i'm worried about the implications of this change -- if schools are only unblocking `https://codeprojects.org` and not `https://*.codeprojects.org`, i'm worried this will break weblab for users midyear

Reply:
Not that it's a complete solution, but could we detect reachability problems when starting up weblab and tell them what they (or their IT teams) need to unblock?

Reply:
we could, i think the only downside of that is it would be quite a nontrivial amount of extra engineering work. i'm not sure how we've handled something like this in the past, but another way to handle this could be: wait to merge this, let teachers know this is happening, give josh some materials to prepare for this change, then merge it over the summer and likely send another reminder to users at the beginning of the school year.

there are likely other ways we could handle this, but that's the only "no extra engineering work" solution i thought of

Reply:
I'd lean towards this being a summer change, would be good to hear Dan's advice

Reply:
We did once switch our youtube URLs.  We first gathered data on reachability [here](https://github.com/code-dot-org/code-dot-org/pull/21885), and then flipped [here](https://github.com/code-dot-org/code-dot-org/pull/22917), and I guess coordinated with various folks on the team to time & communicate this change.  It did have the advantage of having a fallback player which mitigates some risk.

I think it could be a relatively small amount of code to detect non-reachability and show a message about that, but the bigger issue will be of course getting it reachable for all users.

Reply:
thank you for those links, brendan! after our standup today, here's a proposed plan:
- take this to the curriculum sync to get thoughts/concerns from dan and josh
- if we're worried about this hard domain swap (even with a long communication timeline, support articles, etc) causing lots of issues or frustration for users, a possible plan could be:
  -  do the engineering work to gracefully fallback to `downloads.computinginthecore.org`
  - open a dialog for users who fallback and point them to a support article that guides them through this transition. with this, we'll need to set a hard date and communicate that as part of the support/fallback messaging (e.g., "if your IT department hasn't made these changes by 1 Jan 2023, Weblab will no longer load")

Reply:
@sureshc I think that sounds like a fantastic option if its feasible (no change for teachers/IT admins in what they allow). @breville what are your thoughts?

Reply:
This sounds like a good idea to me.

Reply:
hi folks, sorry to butt in.  @sureshc 's proposal sounds good to me.

how will we plan to serve bramble in local development -- will developers need access to S3, or will they be able to fetch bramble from a local server? 

If hitting a local server, my request would be to serve this from dashboard, not pegasus. The reason is that we are likely going to be moving weblab shares on codeprojects.org from pegasus to dashboard soon. if that's complicated for some reason, then I'm happy to help brainstorm on a good path forward.

Reply:
Revisiting this thread as we investigate more reports of Web Lab failing to load successfully.

Firstly, is there any reason we couldn't serve Bramble (currently at https://downloads.computinginthecore.org/bramble_0.1.30) off of a subdomain of https://code.org?  This would remove an entire domain from the [list](https://code.org/educate/it) that teachers need to unblock.

We could start doing it using an optional URL parameter to see if it helps users in the field who otherwise have loading issues, and transition permanently if it seems safe.

Also, with https://github.com/code-dot-org/code-dot-org/pull/49046, we'll hopefully get some more data about domain reachability soon.

Secondly, even local development currently loads from https://downloads.computinginthecore.org/bramble_0.1.30.

Reply:
i can't answer your first question, but as for your second question, we have some awkward "dev mode" code to load bramble locally:

https://github.com/code-dot-org/code-dot-org/blob/2a8e243f8f3a7bf05d086671fda93e43dd783068/dashboard/app/controllers/weblab_host_controller.rb#L8

it defaults to loading from S3 because engineers rarely have a bramble server running locally